include/secp256k1_musig.h

```diff
@@ -197,155 +154,120 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_musig_pubkey_tweak_add(
+ *  4. Avoid copying (or serializing) the secnonce. This reduces the possibility
+ *     that it is used more than once for signing.
  *
- *  Returns: 1: session is successfully initialized
- *           0: session could not be initialized: secret key or secret nonce overflow
+ *  Remember that nonce reuse will immediately leak the secret key!
+ *
+ *  Returns: 0 if the arguments are invalid and 1 otherwise
  *  Args:         ctx: pointer to a context object, initialized for signing (cannot
  *                     be NULL)
- *  Out:      session: the session structure to initialize (cannot be NULL)
- *            signers: an array of signers' data to be initialized. Array length must
- *                     equal to `n_signers` (cannot be NULL)
- * nonce_commitment32: filled with a 32-byte commitment to the generated nonce
- *                     (cannot be NULL)
- *  In:  session_id32: a *unique* 32-byte ID to assign to this session (cannot be
- *                     NULL). If a non-unique session_id32 was given then a partial
- *                     signature will LEAK THE SECRET KEY.
- *              msg32: the 32-byte message to be signed. Shouldn't be NULL unless you
- *                     require sharing nonce commitments before the message is known
- *                     because it reduces nonce misuse resistance. If NULL, must be
- *                     set with `musig_session_get_public_nonce`.
- *        combined_pk: the combined xonly public key of all signers (cannot be NULL)
- *        pre_session: pointer to a musig_pre_session struct after initializing
- *                     it with `musig_pubkey_combine` and optionally provided to
- *                     `musig_pubkey_tweak_add` (cannot be NULL).
- *          n_signers: length of signers array. Number of signers participating in
- *                     the MuSig. Must be greater than 0 and at most 2^32 - 1.
- *           my_index: index of this signer in the signers array. Must be less
- *                     than `n_signers`.
- *             seckey: the signer's 32-byte secret key (cannot be NULL)
+ *  Out:     secnonce: pointer to a structure to store the secret nonce
+ *         pubnonce66: a 66-byte array to store the public nonce
+ *  In:  session_id32: a 32-byte session_id32 as explained above. Must be
+ *                     uniformly random unless you really know what you are
+ *                     doing.
+ *             seckey: the 32-byte secret key that will be used for signing if
+ *                     already known (can be NULL)
+ *              msg32: the 32-byte message that will be signed if already known
+ *                     (can be NULL)
+ *        combined_pk: the combined xonly public key of all signers if already
+ *                     known (can be NULL)
+ *      extra_input32: an optional 32-byte array that is input to the nonce
+ *                     derivation function (can be NULL)
  */
 SECP256K1_API int secp256k1_musig_session_init(
```

Comment:
I think this is not well named. Should be `secp256k1_musig_create_nonce`.

Reply:
Added suggestion to OP.

Reply:
How do you like my suggestion `musig_start` (shorter, simpler)